atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableFactoryRenderer.java

```diff
@@ -51,114 +84,210 @@ public String getClassName() {
+        return definitions.entrySet()
+                .stream()
+                .map(entry -> getTableMethod(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    private MethodSpec getTableMethod(String name, TableDefinition tableDefinition) {
+        String tableName = getTableName(name);
+        String triggerName = tableName + "." + name + "Trigger";
+        TypeName tableType = ClassName.get(packageName, tableName);
+        TypeName triggerType = ClassName.get(packageName, triggerName);
+        MethodSpec.Builder tableGetterMethodBuilder = MethodSpec.methodBuilder("get" + tableName)
+                .addModifiers(Modifier.PUBLIC)
+                .addParameter(Transaction.class, "t")
+                .returns(tableType);
+        if (tableDefinition.getGenericTableName() != null) {
+            tableGetterMethodBuilder
+                    .addParameter(String.class, "name")
+                    .addParameter(ArrayTypeName.of(triggerType), "triggers")
+                    .varargs()
+                    .addStatement("return $T.of(t, namespace, name, $T.getAllTriggers(t, sharedTriggers, triggers))",
+                            tableType,
+                            Triggers.class);
+        } else {
+            tableGetterMethodBuilder
+                    .addParameter(ArrayTypeName.of(triggerType), "triggers")
+                    .varargs()
+                    .addStatement("return $T.of(t, namespace, $T.getAllTriggers(t, sharedTriggers, triggers))",
+                            tableType,
+                            Triggers.class);
+        }
+        return tableGetterMethodBuilder.build();
+    }
+
+    private TypeSpec getSharedTriggers() {
+        TypeSpec.Builder sharedTriggersInterfaceBuilder = TypeSpec.interfaceBuilder("SharedTriggers")
+                .addModifiers(Modifier.PUBLIC);
+
+        for (String name : definitions.keySet()) {
+            String tableName = getTableName(name);
+            String triggerName = tableName + "." + name + "Trigger";
+            TypeName triggerType = ClassName.get(packageName, triggerName);
+            sharedTriggersInterfaceBuilder.addSuperinterface(triggerType);
+        }
+
+        return sharedTriggersInterfaceBuilder.build();
+    }
+
+    private TypeSpec getNullSharedTriggers(TypeName sharedTriggersInterfaceType) {
+        TypeSpec.Builder nullSharedTriggersClassBuilder = TypeSpec.classBuilder("NullSharedTriggers")
+                .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT, Modifier.STATIC)
+                .addSuperinterface(sharedTriggersInterfaceType);
+
+        for (Entry<String, TableDefinition> entry : definitions.entrySet()) {
+            String name = entry.getKey();
+            TableDefinition tableDefinition = entry.getValue();
+            String tableName = getTableName(name);
+            TypeName rowType = ClassName.get(packageName, tableName + "." + name + "Row");
```

Comment:
Instead of using ".", we can probably use the varargs version of this method. As in, instead of the above, the following probably works `ClassName.get(packageName, tableName, name + "Row")`.